AutoCollection/WebSnippet.ts

```diff
@@ -0,0 +1,146 @@
+import * as path from "path";
+import http = require("http");
+import fs = require('fs');
+
+import Logging = require("../Library/Logging");
+import TelemetryClient = require("../Library/TelemetryClient");
+
+class WebSnippet {
+
+    public static INSTANCE: WebSnippet;
+
+    private static _snippet: string;
+    private static _aiUrl: string;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+
+    constructor(client: TelemetryClient) {
+        if (!!WebSnippet.INSTANCE) {
+            throw new Error("Web snippet injection should be configured from the applicationInsights object");
+        }
+
+        WebSnippet.INSTANCE = this;
+        // AI URL used to validate if snippet already included
+        WebSnippet._aiUrl = " https://js.monitor.azure.com/scripts/b/ai.2";
+
+        let snippetPath = path.resolve(__dirname, "../../node_modules/applicationinsight-web-snippet/snippet/snippet.min.js");
+        if (client.config.isDebugWebSnippet) {
+            snippetPath = path.resolve(__dirname, "../../node_modules/applicationinsight-web-snippet/snippet/snippet.js");
+        }
```

Comment:
```suggestion
        let snippetPath = "../../node_modules/applicationinsight-web-snippet/snippet";
        let snippetFile = "snippet.min.js" //default filename
        if (client.config.isDebugWebSnippet) {
            snippetFile = "snippet.js";
        }
        snippetPath = path.resolve(snippetPath, snippetFile); //no need to bring in __dirname since path.resolve will use current folder to resolve as long as your param is a relative folder.
```

Reply:
will update this with npm when snippet is published